### PR TITLE
torch optmize chatglm3

### PR DESCRIPTION
**Pull request by grimoire**
- optimize performance
- chatglm adapter TP support

before
```
concurrency: 256
elapsed_time: 306.702s

first token latency(s)(min, max, ave): 0.281, 10.159, 1.888
per-token latency(s) percentile(50, 75, 95, 99): [0.083, 0.088, 0.271, 0.444]

number of prompt tokens: 724392
number of completion tokens: 662960
token throughput (completion token): 2161.578 token/s
token throughput (prompt + completion token): 4523.455 token/s
RPS (request per second): 9.781 req/s
RPM (request per minute): 586.889 req/min
```

after
```
concurrency: 256
elapsed_time: 261.317s

first token latency(s)(min, max, ave): 0.203, 9.053, 1.658
per-token latency(s) percentile(50, 75, 95, 99): [0.069, 0.077, 0.242, 0.409]

number of prompt tokens: 724392
number of completion tokens: 662960
token throughput (completion token): 2536.999 token/s
token throughput (prompt + completion token): 5309.085 token/s
RPS (request per second): 11.480 req/s
RPM (request per minute): 688.820 req/min
```

### CODE REVIEW

File: lmdeploy/pytorch/config.py [hunk 1]
```diff
@@ -109,14 +110,16 @@ def __build_chatglm():
             bos_token_id = hf_config.bos_token_id
             if bos_token_id is None:
                 bos_token_id = hf_config.pad_token_id
+            init_kwargs = dict(empty_init=False)
```
Comment (RunningLeon) [inline lmdeploy/pytorch/config.py:113]: why need to add `empty_init=False`?

Comment (grimoire) [reply lmdeploy/pytorch/config.py:113]: init_empty_weight in accelerate can not load chatglm to `meta` device with `empty_init=True`, which leads to tp adapter load failed.

Comment (lvhan028) [reply lmdeploy/pytorch/config.py:113]: How to test this model?
The following code failed
```python
pipe = pipeline('/workspace/models-140/chatglm3-6b/', 
                backend_config=PytorchEngineConfig(
                    model_name='base'
                ))
```

Comment (grimoire) [reply lmdeploy/pytorch/config.py:113]: > The following code failed

template failed, fixed in #1168

Comment (lvhan028) [reply lmdeploy/pytorch/config.py:113]: tp=1 pass
tp=2 failed

Comment (lvhan028) [reply lmdeploy/pytorch/config.py:113]: May update FAQs in pipeline.md
The current phenomenon is 
```
raise RuntimeError(f'Rank[{idx}] failed.')
```

Comment (grimoire) [reply lmdeploy/pytorch/config.py:113]: Done

**Review by RunningLeon — APPROVED** [review]
LGTM

**Pull request merged**